ai_nexus_backend/github_api.py

```diff
@@ -6,8 +6,8 @@
 import pandas as pd
 import requests
 from requests.exceptions import HTTPError
-from yaml import safe_load, YAMLError
 
+from ai_nexus_backend.build_yaml import _parse_yaml
 from ai_nexus_backend.requests_utils import (
     _configure_requests,
```

Comment:
I would probably not underscore these if they are imported by other code, even if it is code internal to the library.

Reply:
ah, interesting - this has been very common practice in python code that I've encountered and a practice I've adopted. Signifying internal functions & methods that would not necessarily have the same promise of stability, etc. EG in [PEP8](https://peps.python.org/pep-0008/):

> _single_leading_underscore: weak “internal use” indicator. E.g. from M import * does not import objects whose names start with an underscore.

Not that this is an argument because I would never recommend `from M import *` as a pattern anyway. But anyone attempting it wouldn't get access to the internals. 

I might radically change the implementation of an internal like this and consider it not to be a breaking change. This might be something worth chatting through and aligning on at some point.